### PR TITLE
POC ignore safe canvas calls

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Reviewer:**

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications

### CODE REVIEW

File: shared/js/content-scope/canvas.js [hunk 1]
```diff
@@ -17,38 +17,86 @@ export function computeOffScreenCanvas (canvas, domainKey, sessionKey, getImageD
...
             }
-            // Ignore phaser background
-            if (d[i] === 255 && d[i + 3] === 255) {
-                continue
+
+            ++hits
+            imageData.data[pixelCanvasIndex] = imageData.data[pixelCanvasIndex] ^ (byte & 0x1)
+        })
+
+        if (hits < hitMinimum) {
```
Comment (englehardt) [inline shared/js/content-scope/canvas.js:48]: I wonder whether this is necessary? I suspect that most fingerprinting scripts will simply hash the resulting canvas, so as long as we are modifying some pixels in every canvas the protection should be effective.

Comment (englehardt) [reply shared/js/content-scope/canvas.js:48]: I can see how this is necessary for certain types of canvases. See my comment here: https://app.asana.com/0/0/1200763289112611/1200838871508698/f for more thoughts.

File: shared/js/content-scope/canvas.js [hunk 2]
```diff
@@ -17,38 +17,86 @@ export function computeOffScreenCanvas (canvas, domainKey, sessionKey, getImageD
...
-    // Create an array of only pixels that have data in them
-    const d = imageData.data
-    for (let i = 0; i < d.length; i += 4) {
-        // Blank Blue and Green color
-        if (d[i + 1] === 0 && d[i + 2] === 0) {
-            // Ignore non blank pixels there is high chance compression ignores them
-            if (d[i] === 0 && d[i + 3] === 0) {
-                continue
+    const length = imageData.data.length / 4
+    const hitMinimum = 50
```
[the remark this answers is not in the document]
Comment (jonathanKingston) [reply shared/js/content-scope/canvas.js:21]: No real rationale to these, I was going to make these remote configurable so we can tweak them. I think we could probably decide on a number that is a good trade off on entropy reduction and perf.

**Comment by englehardt** [comment]
I think we can close this out now?

**Pull request closed without merging**